qa/suites/fs/upgrade/featureful_client/old_client/tasks/2-upgrade.yaml

```diff
@@ -21,7 +21,9 @@ tasks:
 - print: "**** done mds pre-upgrade sequence"
 - install.upgrade:
     mon.a:
+      branch: quincy
     mon.b:
+      branch: quincy
```

Comment:
The original code seems buggy, because from `qa/suites/fs/upgrade/featureful_client/old_client/clusters/1-mds-2-client-micro.yaml`: 

```
  1 roles:
  2 - [mon.a, mon.b, mon.c, mgr.x, mgr.y, mds.a, mds.b, mds.c, osd.0, osd.1, osd.2, osd.3]                                                             
  3 - [client.0]
  4 - [client.1]
  5 openstack:
  6 - volumes: # attached to each instance
  7     count: 4
  8     size: 10 # GB
```
The `mon.a` and `mon.b` are exactly the same node. So the `mon.b` here makes no sense ??

Reply:
I would leave that up to teuthology to do the right thing _even_ if the daemons are on the same node. Moreover, if someone changes the yaml in the future to put the monitors of different hosts, then this yaml need not change.

Reply:
I am sure it will run the same upgrade in parallel on the same node, more detail please see https://github.com/ceph/teuthology/blob/main/teuthology/task/install/__init__.py#L78-L83.

It will find the `role` or `remote` from the following config by using `mon.a` and `mon.b`:

```
1 roles:
2 - [mon.a, mon.b, mon.c, mgr.x, mgr.y, mds.a, mds.b, mds.c, osd.0, osd.1, osd.2, osd.3] 
3 - [client.0] 
4 - [client.1]
```
While all the `mon/mds/mgr/mds/osd` daemons are on the same node.

But for the rpm packages the rpm will help lock and won't mess it. So it should be okay.

Reply:
yeh - it does the right thing :)